### PR TITLE
Update documentation for dbt-sqlserver and dbt-synapse

### PR DESCRIPTION
**Pull request by sdebruyn**
## Description & motivation
Updating the documentation for the dbt-sqlserver and dbt-synapse adapters since the information was not up to date anymore

- [x] Updated profile info for dbt-sqlserver
- [x] Updated profile info for dbt-synapse
- [x] Updated configuration options for dbt-sqlserver
- [x] Updated configuration options for dbt-synapse

## To-do before merge
/

## Prerelease docs
/

## Checklist
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

### CODE REVIEW

**Comment by sdebruyn** [comment]
The goal is - once this gets merged - that we remove all the extra information we have in our repo's READMEs and just point to the dbt docs. Then we'd only have to maintain these things in 1 place.

File: website/docs/reference/warehouse-profiles/azuresynapse-profile.md [hunk 1]
```diff
@@ -12,167 +11,62 @@ Some core functionality may be limited. If you're interested in contributing, ch
...
-</File>
-
-</TabItem>
 
-<TabItem value="interactive">
+:::info Dedicated SQL only
 
-*Windows Only* brings up the Azure AD prompt so you can MFA if need be.
+Azure Synapse offers both Dedicated SQL Pools and Serverless SQL Pools.
+**Only Dedicated SQL Pools are supported by this adapter.**
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/azuresynapse-profile.md]: might be worth linking this throwaway project? https://github.com/dbt-msft/dbt-synapse-serverless
```suggestion
**Only Dedicated SQL Pools are supported by this adapter. If you really insist on using serverless pools, check out the neglected, experimental project: [dbt-synapse-serverless](https://github.com/dbt-msft/dbt-synapse-serverless)** 
```

File: website/docs/reference/warehouse-profiles/azuresynapse-profile.md [hunk 2]
```diff
@@ -12,167 +11,62 @@ Some core functionality may be limited. If you're interested in contributing, ch
...
+      user: username
+      password: password
 ```
 
 </File>
 
-</TabItem>
-
-</Tabs>
+You can find all the available options and the documentation and how to configure them on [the documentation page for the dbt-sqlserver adapter](mssql-profile#connecting-to-azure-sql-server-with-dbt-sqlserver).
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/azuresynapse-profile.md:72]: love the simplification here!

File: website/docs/reference/warehouse-profiles/mssql-profile.md [hunk 3]
```diff
@@ -11,83 +11,135 @@ Some core functionality may be limited. If you're interested in contributing, ch
...
+    {label: 'Windows credentials', value: 'windows'}
+  ]}
+>
+
+<TabItem value="password">
 
 <File name='profiles.yml'>
 
-```yml
+```yaml
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mssql-profile.md:58]: lol

File: website/docs/reference/warehouse-profiles/mssql-profile.md [hunk 4]
```diff
@@ -11,83 +11,135 @@ Some core functionality may be limited. If you're interested in contributing, ch
...
+- Azure CLI authentication
+- VS Code authentication (available through the automatic option below)
+- Azure PowerShell module authentication (available through the automatic option below)
+- Automatic authentication
+
+The automatic authentication setting is in most cases the easiest choice and works for all of the above.
 
 <Tabs
-  defaultValue="integrated"
+  defaultValue="aad_password"
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mssql-profile.md]: I get why integrated shouldn't be listed, but should this be the default?
```suggestion
  defaultValue="azure_cli"
```

File: website/docs/reference/warehouse-profiles/mssql-profile.md [hunk 5]
```diff
@@ -97,106 +149,200 @@ your_profile_name:
...
-This is also the preferred route for using a service principal:
-
-`az login --service-principal --username $CLIENTID --password $SECRET --tenant $TENANTID`
 
 </File>
 
 </TabItem>
 
-<TabItem value="interactive">
+<TabItem value="auto">
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mssql-profile.md:252]: I didn't even know this existed?!? lol

File: website/sidebars.js [hunk 6]
```diff
@@ -307,6 +307,8 @@ const sidebarSettings = {
         "reference/resource-configs/materialize-configs",
         "reference/resource-configs/firebolt-configs",
         "reference/resource-configs/teradata-configs",
+        "reference/resource-configs/mssql-configs",
```
Comment (dataders) [inline website/sidebars.js:310]: weird that this didn't exist before in the sidebar...

**Review by dataders — APPROVED** [review]
thanks @sdebruyn! I only had small suggestions. This overhaul is much appreciated!

File: website/docs/reference/warehouse-profiles/mssql-profile.md (hunk 5 above)
Comment (sdebruyn) [reply website/docs/reference/warehouse-profiles/mssql-profile.md:252]: This the one I'd recommend most. Just login with anything and it will work.

**Comment by sdebruyn** [comment]
Fixed the remarks, thanks for the review @dataders :)

**Review by dataders — APPROVED** [review]
🚢

**Pull request merged**